File: qc-application/src/main/java/com/eager/questioncloud/application/business/payment/question/service/QuestionPaymentHistoryService.java
package com.eager.questioncloud.application.business.payment.question.service;

import com.eager.questioncloud.core.common.PagingInformation;
import com.eager.questioncloud.core.domain.payment.infrastructure.repository.QuestionPaymentHistoryRepository;
import com.eager.questioncloud.core.domain.payment.model.QuestionPaymentHistory;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class QuestionPaymentHistoryService {
    private final QuestionPaymentHistoryRepository questionPaymentHistoryRepository;

    public List<QuestionPaymentHistory> getQuestionPaymentHistory(Long userId, PagingInformation pagingInformation) {
        return questionPaymentHistoryRepository.getQuestionPaymentHistory(userId, pagingInformation);
    }

    public int countQuestionPaymentHistory(Long userId) {
        return questionPaymentHistoryRepository.count(userId);
    }
}

File: qc-application/src/main/java/com/eager/questioncloud/application/business/payment/question/service/QuestionPaymentHistoryService.kt
package com.eager.questioncloud.application.business.payment.question.service

import com.eager.questioncloud.core.common.PagingInformation
import com.eager.questioncloud.core.domain.payment.infrastructure.repository.QuestionPaymentHistoryRepository
import com.eager.questioncloud.core.domain.payment.model.QuestionPaymentHistory
import org.springframework.stereotype.Service

@Service
class QuestionPaymentHistoryService(
    private val questionPaymentHistoryRepository: QuestionPaymentHistoryRepository
) {
    fun getQuestionPaymentHistory(userId: Long, pagingInformation: PagingInformation): List<QuestionPaymentHistory> {
        return questionPaymentHistoryRepository.getQuestionPaymentHistory(userId, pagingInformation)
    }

    fun countQuestionPaymentHistory(userId: Long): Int {
        return questionPaymentHistoryRepository.count(userId)
    }
}
